numpyro/mcmc.py

```diff
@@ -688,7 +688,9 @@ def _single_chain_mcmc(self, init, collect_fields=('z',), collect_warmup=False,
         hmc_state, constrain_fn = self.sampler.init(rng, self.num_warmup, init_params,
                                                     model_args=args, model_kwargs=kwargs)
         if self.constrain_fn is None:
-            self.constrain_fn = identity if constrain_fn is None else constrain_fn
+            constrain_fn = identity if constrain_fn is None else constrain_fn
+        else:
+            constrain_fn = self.constrain_fn
```

Comment:
We won't use `constrain_fn` elsewhere, so it is better to not set this property here. The advantage, as shown in bayesian regression tutorial, is that users don't have to redefine `kernel`, `mcmc` to get samples if args in `run()` method changes.